### PR TITLE
Fix some ANSI-C issues

### PR DESCRIPTION
**Pull request by pisculichi**

### CODE REVIEW

**Comment by dequis** [comment]
Thanks but all of those are perfectly valid c99. Do you have a specific example of compiler/platform that doesn't support this? Because we don't support microsoft visual studio.

**Comment by pisculichi** [comment]
it's for Haiku-os gcc2 (https://github.com/haikuports/haikuports/pull/553)

**Comment by ailin-nemui** [comment]
Are you sure those casts are required?

**Comment by pisculichi** [comment]
@ailin-nemui honestly no, I'll check if are required.

File: src/fe-text/gui-entry.c [hunk 1]
```diff
@@ -582,12 +583,14 @@ char *gui_entry_get_cutbuffer(GUI_ENTRY_REC *entry)
 
 char *gui_entry_get_next_cutbuffer(GUI_ENTRY_REC *entry)
 {
+	GUI_ENTRY_CUTBUFFER_REC *tmp = NULL;
```
Comment (ailin-nemui) [inline src/fe-text/gui-entry.c]: = NULL not necessary here

File: src/fe-text/gui-entry.c [hunk 2]
```diff
@@ -652,7 +655,8 @@ void gui_entry_erase(GUI_ENTRY_REC *entry, int size, CUTBUFFER_UPDATE_OP update_
 	if (update_cutbuffer != CUTBUFFER_UPDATE_NOOP) {
 		int cutbuffer_new_size;
 		unichar *tmpcutbuffer;
-		GUI_ENTRY_CUTBUFFER_REC *tmp = get_cutbuffer_rec(entry, update_cutbuffer);
+		GUI_ENTRY_CUTBUFFER_REC *tmp;
```
Comment (ailin-nemui) [inline src/fe-text/gui-entry.c]: this change is not necessary as far as I can tell

File: src/fe-text/gui-readline.c [hunk 3]
```diff
@@ -696,10 +696,11 @@ static void key_append_next_kill(void)
 
 static gboolean paste_timeout(gpointer data)
 {
+	int split_lines = 0;
```
Comment (ailin-nemui) [inline src/fe-text/gui-readline.c]: this initialisation is also not required

**Comment by pisculichi** [comment]
Fixed!

**Pull request merged**